web-ui/src/main/resources/catalog/components/common/ows/OWSService.js

```diff
@@ -443,69 +443,82 @@
 
-            //non namespaced lowercase name
-            nameNoNamespace = name.split(':')[
-                name.split(':').length - 1].toLowerCase();
+            // Layer name may be a list of comma separated layers
+            layerList = layerName.split(',');
+            for (var j = 0; j < layerList.length; j ++) {
+              var name = layerList[j];
 
-            for (var i = 0; i < layers.length; i++) {
-              //Add Info for Requests:
-              if (capObj.Request) {
-                layers[i].capRequest = capObj.Request;
-              }
+              //non namespaced lowercase name
+              nameNoNamespace = name.split(':')[
+                  name.split(':').length - 1].toLowerCase();
 
-              //check layername
-              var lId = layers[i].Identifier;
-              var capName = layers[i].Name ||
-                  (lId && angular.isArray(lId) ? lId[0] : lId) || '',
-                  capNameNoNamespace;
-              //non namespaced lowercase capabilities name
-              if (capName) {
-                capNameNoNamespace = capName.split(':')[
-                    capName.split(':').length - 1].toLowerCase();
-              }
+              for (var i = 0; i < layers.length; i++) {
+                //Add Info for Requests:
+                if (capObj.Request) {
+                  layers[i].capRequest = capObj.Request;
+                }
 
-              //either names match or non namespaced names
-              if (name == capName || nameNoNamespace == capNameNoNamespace) {
-                layers[i].nameToUse = capName;
-                if (capObj.version) {
-                  layers[i].version = capObj.version;
+                //check layername
+                var lId = layers[i].Identifier;
+                var capName = layers[i].Name ||
+                    (lId && angular.isArray(lId) ? lId[0] : lId) || '',
+                    capNameNoNamespace;
+                //non namespaced lowercase capabilities name
+                if (capName) {
+                  capNameNoNamespace = capName.split(':')[
+                      capName.split(':').length - 1].toLowerCase();
                 }
-                return layers[i];
-              }
 
-              //check dataset identifer match
-              if (uuid != null) {
-                if (angular.isArray(layers[i].Identifier)) {
-                  angular.forEach(layers[i].Identifier, function(id) {
-                    if (id == uuid) {
-                      needles.push(layers[i]);
-                    }
-                  });
+                //either names match or non namespaced names
+                if (name == capName || nameNoNamespace == capNameNoNamespace) {
+                  layers[i].nameToUse = capName;
+                  if (capObj.version) {
+                    layers[i].version = capObj.version;
+                  }
+                  needles.push(layers[i]);
+                  continue;
```

Comment:
This should not be better a `break` instead of `continue`? If there's a match, should not continue processing the rest of capabilities layers, no?

Similar comment in other loops below using `continue` when there's a match.

Apart from this, I've tested and works fine.

Reply:
No because you want to check that all layers are present, and as we match on layer name, then layer identifier, then layer metadata url, we have to continue.

Reply:
Yes you're right

Reply:
Break proper loop once found now.